src/sentry/models/group.py

```diff
@@ -45,6 +54,28 @@ class GroupStatus(object):
 class GroupManager(BaseManager):
     use_for_related_fields = True
 
+    def by_qualified_short_id(self, org, short_id):
+        match = _short_id_re.match(short_id.strip())
+        if match is None:
+            return
```

Comment:
Thinking about it more, performing the validation in this method at all seems confusing (why would someone be passing invalid inputs on purpose to this method in the first place?) and that it should probably get the project callsign and issue short ID already split out.